6_Sviatlana_Yoshchyk.py

```diff
@@ -0,0 +1,155 @@
+
+# for return sub-vocabulary with given user 
+def find_name(struct, key):
+    if key in struct:
+        return struct[key]
+    for el_struct in struct.values():
+        if isinstance(el_struct, dict):
+            result=find_name(el_struct, key)
+            if result:
+                break
+    else:
+        result = None
+    return result
+        
+while True:
+# clear screen    
+    os.system('cls')
+    print("Menu")
+    print("\n1. Display a list of users \n2. View user information \n3. Change user details \n4. Delete selected user \n5. Add user \n6. Exit")
+    answer = int(input("Choose an action: "))
+    os.system('cls')
+    if  answer == 1:
+        print("List of users: ")
+        print( )
+        list_name()        
+
+    elif  answer == 2:
+        print("List of users: ")
+        print( )
+        list_name()
+        answre_info_name = input("Who do you need information about? Enter name: ")
+        os.system('cls')
+        list_data = find_name(data_cal_bme, answre_info_name)
+        if list_data:
+            height_name = list_data["height"]
+            weight_name = list_data["weight"]   
+            print("Info about", answre_info_name)   
+            print("height:", height_name)
+            print("weight:", weight_name)            
+            print(cal_bme(height_name, weight_name))   
+
+        else:
+            print("There is no such person")
+
+    elif  answer == 3:
+        print("List of users: ")
+        print( )
+        list_name()
+        answer_name = input("Which user do you want to change information about? ")
+        os.system('cls')
+        print(f"What information about {answer_name} would you like to change?") 
+        print("1. Height \n2. Weight \n3. Gender \n4. Age")
+        choice = int(input("Make a choice: "))
+        os.system('cls')
+        list_data = find_name(data_cal_bme, answer_name)
+        if list_data:
+            if choice == 1:
+                new_height = float(input("Enter his height: "))
+                find_name(data_cal_bme, answer_name)["height"] = new_height
+                print(data_cal_bme)
+            elif choice == 2:
+                new_weight=int(input("Enter his weight: "))
+                find_name(data_cal_bme, answer_name)["weight"] = new_weight
+                print(data_cal_bme)
+            elif choice == 3:
+                new_gender=input("Enter his gender (female or male): ")
+                find_name(data_cal_bme, answer_name)["gender"] = new_gender
+                print(data_cal_bme)
+            elif choice == 4:
+                new_age= int(input("Enter his age: "))
+                find_name(data_cal_bme, answer_name)["age"] = new_age
+                print(data_cal_bme)
+        else:
+            print("There is no such person")
+            
+    elif  answer == 4:
+        print("List of users: ")
+        list_name()
+        answre_info_name = input("Which user do you want to remove? ")
```

Comment:
If the users have IDs in the dict I would just DEL/UPDATE them by ID not by name.
The logic would be easy.
Moreover, user names could overlap..